camel-core/src/main/java/org/apache/camel/component/seda/SedaProducer.java

```diff
@@ -236,13 +239,24 @@ protected void addToQueue(Exchange exchange, boolean copy) throws SedaConsumerNo
         }
 
         log.trace("Adding Exchange to queue: {}", target);
-        if (blockWhenFull) {
+        if (blockWhenFull && offerTimeout == 0) {
             try {
                 queue.put(target);
             } catch (InterruptedException e) {
                 // ignore
                 log.debug("Put interrupted, are we stopping? {}", isStopping() || isStopped());
             }
+        } else if (blockWhenFull && offerTimeout > 0) {
+            try {
+                offerTime = queue.offer(target, offerTimeout, TimeUnit.MILLISECONDS);
+                if (!offerTime) {
+                    throw new IllegalStateException(" Fails to Insert the specified element into this queue, "
```

Comment:
Remove leading whitespace. Also maybe improve the text a little bit such as `Failed to insert element into queue after timeout of X milliseconds`

Reply:
Btw the JDK does just this: 

    throw new IllegalStateException("Queue full");